tests/Doctrine/Tests/Common/Cache/CacheTest.php

```diff
@@ -29,6 +29,21 @@ public function testBasicCrudOperations($value)
         $this->assertFalse($cache->contains('key'));
     }
 
+    public function testCacheKeyIsCaseSensitive()
+    {
+        $cache = $this->_getCacheDriver();
+
+        $this->assertTrue($cache->save('key', 'value'));
+        $this->assertTrue($cache->contains('key'));
+        $this->assertSame('value', $cache->fetch('key'));
+
+        $this->assertFalse($cache->contains('KEY'));
+        $this->assertFalse($cache->fetch('KEY'));
+
+        $cache->delete('KEY');
+        $this->assertTrue($cache->contains('key', 'Deleting cache item with different case must not affect other cache item'));
```

Comment:
> I don't know a way to find this collision

Same issue here, but collisions are really dangerous here.
I'd be happy to have a filesystem test that verifies that the file is saved in a specific location for two case-differing keys. This test case can still live here, as it serves other adapters as well